tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -575,7 +501,7 @@ static PyObject * THPVariable_numel(PyObject* self_, PyObject* args, PyObject* k
   auto r = parser.parse(args, kwargs, parsed_args);
 
   if(r.has_torch_function()){
-    return handle_torch_function(r, args, kwargs, THPVariableFunctions);
+    return handle_torch_function(r, args, kwargs, (PyObject*)&THPVariableFunctions, "torch");
```

Comment:
What effectively is an unchecked `reinterpret_cast` is kind of annoying here. I wonder if you can just write a little helper function that gives you THPVariableFunctions at the right type, so that there's less chance for things to go wrong.